modules/citrus-integration/src/it/java/com/consol/citrus/javadsl/design/WaitJavaIT.java

```diff
@@ -51,19 +49,80 @@ public void waitFile() throws IOException {
     }
 
     @CitrusTest
-    public void waitHttp() {
+    public void waitHttpDeprecated() {
+
+        //GIVEN
+        String server = String.format("http://localhost:%s", serverPort);
         httpServer.setPort(serverPort);
-        httpServer.setEndpointAdapter(new EmptyResponseEndpointAdapter());
+        start(httpServer);
+
+        parallel().actions(
+                sequential().actions(
+                        //WHEN
+                        waitFor()
+                                .http(server).getBuilder().build()
+                ),
+                sequential().actions(
+                        //THEN
+                        http().server(httpServer).receive().head(),
+                        http().server(httpServer).respond(HttpStatus.NOT_FOUND),
+                        http().server(httpServer).receive().head(),
+                        http().server(httpServer).respond(HttpStatus.OK)
+                )
+        );
 
+        doFinally().actions(stop(httpServer));
+    }
+
+    @CitrusTest
+    public void waitHttpAsAction() {
+
+        //GIVEN
+        String server = String.format("http://localhost:%s", serverPort);
+        httpServer.setPort(serverPort);
         start(httpServer);
 
-        waitFor()
-            .http()
-            .url(String.format("http://localhost:%s", serverPort));
+        parallel().actions(
+                sequential().actions(
+                        //WHEN
+                        waitFor()
+                                .execution()
+                                .action(send(server))
```

Comment:
Well now as I look at the `waitAction()` test, I don't think, that it tests anything at all from a functional perspective.
The test ensures that the types are matching with the builder and that the API is well formed but it does not ensure the the `waitFor()` command really waits for anything. Maybe I miss something here.

Nevertheless, `waitHttpAsAction` is an integration test of the HTTP component with the wait for container. And this test already revealed a inconsistency between
```java
waitFor()
  .execution()
  .action(send(server))
```
and 
```java
waitFor()
  .http()
  .url(server)
```

The first option sends `POST` request while the second sends `HEAD` requests. I've opened #564 for this.

So from my perspective, yes, there is a redundancy, because the wait container is tested twice but the citrus functionality that integrates with the container is different what makes it a different test case.in my opinion

Reply:
Clarified via phone. Summary:

- We're currently missing some unit tests here, therefore, we'll keep the tests